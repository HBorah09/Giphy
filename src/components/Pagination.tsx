import { FC } from 'react';
import { Pagination as BootStrapPagination }  from 'react-bootstrap'

interface Props {
    currentPage: number,
    handlePagination: (pageNum: number) => void,
}

const Pagination:FC <Props> = ({ currentPage, handlePagination }) => {
    const pageNumbers = [currentPage-2, currentPage-1, currentPage, currentPage+1, currentPage+2]
    return(
        <BootStrapPagination className='pagination'>
            <BootStrapPagination.Prev disabled={currentPage === 1} onClick={() => handlePagination(currentPage-1)} />
            {pageNumbers.map(num => (
                <div key={`page-${num}`}>
                    {num >0 && <BootStrapPagination.Item 
                        active={ num === currentPage }
                        onClick={ () => handlePagination(num) }
                    >
                        {num}
                    </BootStrapPagination.Item>}
                </div>
            ))}
            <BootStrapPagination.Next onClick={() => handlePagination(currentPage+1)}  />
        </BootStrapPagination>
    )
}

export default Pagination;