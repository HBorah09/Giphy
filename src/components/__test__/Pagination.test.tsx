import { shallow, mount } from 'enzyme';
import { Pagination as BootStrapPagination }  from 'react-bootstrap';
import Pagination from '../Pagination';

describe('<Pagination />', () => {

    const props = {
       currentPage: 3,
       handlePagination: () => jest.fn()
    }

    it('renders without crashing', () => {
        const wrapper = shallow(<Pagination { ...props } />);
        expect(wrapper).toMatchSnapshot();
    });
    
    // it('should call handlePagination with currentPage-1 when PrevButton is clicked', () => {
    //     const wrapper = shallow(<Pagination { ...props } />);
    //     const prevButton = wrapper.find(BootStrapPagination.Prev);
    //     const spy = jest.spyOn(wrapper.instance(), "handlePagination");
    //     prevButton.simulate('click');
    //     expect(spy).toHaveBeenCalledWith(2);
    // });
})