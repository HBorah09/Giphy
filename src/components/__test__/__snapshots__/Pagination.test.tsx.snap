// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders without crashing 1`] = `
<Pagination
  className="pagination"
>
  <Prev
    disabled={false}
    onClick={[Function]}
  />
  <div
    key="page-1"
  >
    <PageItem
      active={false}
      activeLabel="(current)"
      disabled={false}
      onClick={[Function]}
    >
      1
    </PageItem>
  </div>
  <div
    key="page-2"
  >
    <PageItem
      active={false}
      activeLabel="(current)"
      disabled={false}
      onClick={[Function]}
    >
      2
    </PageItem>
  </div>
  <div
    key="page-3"
  >
    <PageItem
      active={true}
      activeLabel="(current)"
      disabled={false}
      onClick={[Function]}
    >
      3
    </PageItem>
  </div>
  <div
    key="page-4"
  >
    <PageItem
      active={false}
      activeLabel="(current)"
      disabled={false}
      onClick={[Function]}
    >
      4
    </PageItem>
  </div>
  <div
    key="page-5"
  >
    <PageItem
      active={false}
      activeLabel="(current)"
      disabled={false}
      onClick={[Function]}
    >
      5
    </PageItem>
  </div>
  <Next
    onClick={[Function]}
  />
</Pagination>
`;
